From ad5f491a9910553f2d71b8a9056816d964396e9a Mon Sep 17 00:00:00 2001
From: Kaviarasu Sanjeevi <quic_kaviaras@quicinc.com>
Date: Tue, 2 Apr 2024 11:37:54 +0530
Subject: [PATCH] ath11k: Compilation fixes for Backports 6.6

1. strlcpy was deprecated in kernel-6.6.3, replaced with strscp,
2. vm_flags_set API was used to assign vm_flags in kernel-6.6.3.
3. Added GFP_KERNEL param for iommu_map API

Signed-off-by: Kaviarasu Sanjeevi <quic_kaviaras@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/ahb.c	    |  4 ++--
 drivers/net/wireless/ath/ath11k/coredump.c |  2 +-
 drivers/net/wireless/ath/ath11k/pktlog.c   |  4 ++++
 3 files changed, 7 insertions(+), 3 deletion(-)

--- a/drivers/net/wireless/ath/ath11k/coredump.c
+++ b/drivers/net/wireless/ath/ath11k/coredump.c
@@ -115,7 +115,7 @@ void ath11k_coredump_build_inline(struct
 	ATH11K_MEMORY_STATS_INC(ab, malloc_size, header_size);
 
 	file_data = (struct ath11k_dump_file_data *)buf;
-	strlcpy(file_data->df_magic, "ATH11K-FW-DUMP",
+	strscpy(file_data->df_magic, "ATH11K-FW-DUMP",
 	        sizeof(file_data->df_magic));
 	file_data->len = cpu_to_le32(header_size);
 	file_data->version = cpu_to_le32(ATH11K_FW_CRASH_DUMP_VERSION);
--- a/drivers/net/wireless/ath/ath11k/pktlog.c
+++ b/drivers/net/wireless/ath/ath11k/pktlog.c
@@ -195,7 +195,11 @@ static int ath_pktlog_mmap(struct file *
 		return -ENOMEM;
 	}
 
+#if LINUX_VERSION_IS_LESS(6, 6, 3)
 	vma->vm_flags |= VM_LOCKED;
+#else
+	vm_flags_set(vma, VM_LOCKED);
+#endif
 	vma->vm_ops = &pktlog_vmops;
 
 	return 0;
--- a/drivers/net/wireless/ath/ath11k/ahb.c
+++ b/drivers/net/wireless/ath/ath11k/ahb.c
@@ -1137,7 +1137,7 @@ static int ath11k_ahb_fw_resources_init(
 
 	ret = iommu_map(iommu_dom, ab_ahb->fw.msa_paddr,
 			ab_ahb->fw.msa_paddr, ab_ahb->fw.msa_size,
-			IOMMU_READ | IOMMU_WRITE);
+			IOMMU_READ | IOMMU_WRITE, GFP_KERNEL);
 	if (ret) {
 		ath11k_err(ab, "failed to map firmware region: %d\n", ret);
 		goto err_iommu_detach;
@@ -1145,7 +1145,7 @@ static int ath11k_ahb_fw_resources_init(
 
 	ret = iommu_map(iommu_dom, ab_ahb->fw.ce_paddr,
 			ab_ahb->fw.ce_paddr, ab_ahb->fw.ce_size,
-			IOMMU_READ | IOMMU_WRITE);
+			IOMMU_READ | IOMMU_WRITE, GFP_KERNEL);
 	if (ret) {
 		ath11k_err(ab, "failed to map firmware CE region: %d\n", ret);
 		goto err_iommu_unmap;
